specs/app.spec.js

```diff
@@ -0,0 +1,149 @@
+const fs = require('fs'),
+    mysql = require('mysql'),
+    parse = require('csv-parse'),
+    expect = require('chai').expect,
+    DB_CONFIG = require('../dbconfig.json'),
+    sinon = require('sinon');
+
+describe('When the user has csv files in the files folder', function () {
+    this.timeout(10000);
+
+    beforeEach(function () {
+        return ensureTablesRemoved(['t1', 't2', 'tr']);
+    });
+
+    afterEach(function () {
+        return ensureTablesRemoved(['t1', 't2', 'tr']);
+    });
+
+    describe('And runs the app.js from the command line with no options', function () {
+        beforeEach(() => {
+            return runCommand('node app.js');
+        });
+
+        it('Should complete without error', function () {
+        });
+
+        it('Should have inserted the t1 csv as expected', function (done) {
+            var connection = mysql.createConnection(DB_CONFIG);
+            connection.query('SELECT * FROM t1', (error, data) => {
+                connection.end();
+                if (error) {
+                    done(error);
+                }
+                expect(data).to.deep.equal([
+                    { '"id"': "1", 'first name': '"SMH"', "person's email": 'smh@email.com' },
+                    { '"id"': "2", 'first name': "MO'", "person's email": 'mo@email.com' },
+                    { '"id"': "3", 'first name': 'hello, world', "person's email": 'someemail@email.com' }
+                ]);
+                done();
+            });
+        });
+
+        it('Should have inserted the t2 csv as expected', function (done) {
+            var connection = mysql.createConnection(DB_CONFIG);
+            connection.query('SELECT * FROM t2', (error, data) => {
+                connection.end();
+                if (error) {
+                    done(error);
+                }
+                expect(data).to.deep.equal([
+                    { 'COL1': '"Hello, World!"', 'COL2': '```' }
+                ]);
+                done();
+            });
+        });
+
+        it('should end the connection after finishing inserting the csv files', function () {
+
+        });
+    });
+
+    describe('When the user runs app.js using the --files flag', function () {
+
+        beforeEach(function () {
+            return runCommand('node app.js --files tr.csv');
```

Comment:
Please include tr.csv in source control. It would be good to put it inside a folder as well and rename it to something like `test-1.csv`. Maybe a folder called `data` under `specs`? Then you could do `node app.js --files specs/data/test-1.csv`.